admin/commands/management/add_mongodb.go

```diff
@@ -179,16 +184,17 @@ func (cmd *AddMongoDBCommand) RunCmd() (commands.Result, error) {
 				QANMongodbProfiler: cmd.QuerySource == MongodbQuerySourceProfiler,
 				QANMongodbMongolog: cmd.QuerySource == MongodbQuerySourceMongolog,
 
-				CustomLabels:                  customLabels,
-				SkipConnectionCheck:           cmd.SkipConnectionCheck,
-				MaxQueryLength:                cmd.MaxQueryLength,
-				TLS:                           cmd.TLS,
-				TLSSkipVerify:                 cmd.TLSSkipVerify,
-				TLSCertificateKey:             tlsCertificateKey,
-				TLSCertificateKeyFilePassword: cmd.TLSCertificateKeyFilePassword,
-				TLSCa:                         tlsCa,
-				AuthenticationMechanism:       cmd.AuthenticationMechanism,
-				AuthenticationDatabase:        cmd.AuthenticationDatabase,
+				CustomLabels:                   customLabels,
+				SharedEnvironmentVariableNames: agentVarNames,
```

Comment:
```suggestion
				EnvironmentVariableNames: agentVarNames,
```
It can be made shorter, envars are mostly shared by definition.